receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -163,52 +179,117 @@ func (k *K8sAPIServer) GetMetrics() []pmetric.Metrics {
-			"namespace_number_of_running_pods": podNum,
+			ci.MetricName(ci.TypeClusterDeployment, ci.SpecReplicas):              deployment.Spec.Replicas,              // deployment_spec_replicas
+			ci.MetricName(ci.TypeClusterDeployment, ci.StatusReplicas):            deployment.Status.Replicas,            // deployment_status_replicas
+			ci.MetricName(ci.TypeClusterDeployment, ci.StatusReplicasAvailable):   deployment.Status.AvailableReplicas,   // deployment_status_replicas_available
+			ci.MetricName(ci.TypeClusterDeployment, ci.StatusReplicasUnavailable): deployment.Status.UnavailableReplicas, // deployment_status_replicas_unavailable
 		}
 		attributes := map[string]string{
 			ci.ClusterNameKey: clusterName,
-			ci.MetricType:     ci.TypeClusterNamespace,
+			ci.MetricType:     ci.TypeClusterDeployment,
 			ci.Timestamp:      timestampNs,
-			ci.K8sNamespace:   namespace,
+			ci.PodNameKey:     deployment.Name,
+			ci.K8sNamespace:   deployment.Namespace,
 			ci.Version:        "0",
 		}
 		if k.nodeName != "" {
-			attributes["NodeName"] = k.nodeName
+			attributes[ci.NodeNameKey] = k.nodeName
 		}
 		attributes[ci.SourcesKey] = "[\"apiserver\"]"
-		attributes[ci.Kubernetes] = fmt.Sprintf("{\"namespace_name\":\"%s\"}", namespace)
+		//attributes[ci.Kubernetes] = fmt.Sprintf("{\"namespace_name\":\"%s\",\"deployment_name\":\"%s\"}",
+		//	deployment.Namespace, deployment.Name)
 		md := ci.ConvertToOTLPMetrics(fields, attributes, k.logger)
-		result = append(result, md)
+		metrics = append(metrics, md)
 	}
+	return metrics
+}
 
-	return result
+func (k *K8sAPIServer) getDaemonSetMetrics(clusterName, timestampNs string) []pmetric.Metrics {
+	var metrics []pmetric.Metrics
+	daemonSets := k.daemonSetClient.DaemonSetInfos()
+	for _, daemonSet := range daemonSets {
+		fields := map[string]interface{}{
+			ci.MetricName(ci.TypeClusterDaemonSet, ci.StatusNumberAvailable):        daemonSet.Status.NumberAvailable,        // daemonset_status_number_available
+			ci.MetricName(ci.TypeClusterDaemonSet, ci.StatusNumberUnavailable):      daemonSet.Status.NumberUnavailable,      // daemonset_status_number_unavailable
+			ci.MetricName(ci.TypeClusterDaemonSet, ci.StatusDesiredNumberScheduled): daemonSet.Status.DesiredNumberScheduled, // daemonset_status_desired_number_scheduled
+			ci.MetricName(ci.TypeClusterDaemonSet, ci.StatusCurrentNumberScheduled): daemonSet.Status.CurrentNumberScheduled, // daemonset_status_current_number_scheduled
+		}
+		attributes := map[string]string{
+			ci.ClusterNameKey: clusterName,
+			ci.MetricType:     ci.TypeClusterDaemonSet,
+			ci.Timestamp:      timestampNs,
+			ci.PodNameKey:     daemonSet.Name,
+			ci.K8sNamespace:   daemonSet.Namespace,
+			ci.Version:        "0",
+		}
+		if k.nodeName != "" {
+			attributes[ci.NodeNameKey] = k.nodeName
+		}
+		attributes[ci.SourcesKey] = "[\"apiserver\"]"
+		//attributes[ci.Kubernetes] = fmt.Sprintf("{\"namespace_name\":\"%s\",\"daemonset_name\":\"%s\"}",
```

Comment:
Commented out for now as Im not sure if adding these will break Container Insights console. Will uncomment in a follow up PR once I confirm this is not a breaking change for them.

Reply:
Oh interesting. Maybe a question for Robert Ramirez Vique